jdk/src/share/bin/java.c

```diff
@@ -297,29 +335,18 @@ JLI_Launch(int argc, char ** argv,              /* main argc, argc */
         SetClassPath(cpath);
     }
 
+    /* Process -Xmso to set the main thread stack size. May be overridden by
+     * a later command line option.
+     */
     {
-        /* Process -Xmso in the OPENJ9_JAVA_OPTIONS environment variable to
-         * set the main thread stack size. May be overridden by a later command
-         * line option.
-         */
-        JLI_List openj9Args = JLI_List_new(8); /* 8 is arbitrary */
-        if (JLI_ParseOpenJ9ArgsFromEnvVar(openj9Args, "OPENJ9_JAVA_OPTIONS")) {
-            size_t i = openj9Args->size;
-            while (i > 0) {
-                i -= 1;
-                if (JLI_StrCCmp(openj9Args->elements[i], "-Xmso") == 0) {
-                    jlong tmp = 0;
-                    if (parse_size(openj9Args->elements[i] + 5, &tmp)) {
-                        threadStackSize = tmp;
-                        if (threadStackSize > 0 && threadStackSize < (jlong)STACK_SIZE_MINIMUM) {
-                            threadStackSize = STACK_SIZE_MINIMUM;
-                        }
-                    }
-                    break;
-                }
-            }
-            JLI_List_free(openj9Args);
-        }
+#define LIB_PATH "/lib/"
+#define OPTIONS_DEFAULT_PATH "/default/options.default"
+        char optionsfile[sizeof(jrepath) + sizeof(LIB_PATH) - 1 + sizeof(LIBARCHNAME) - 1 + sizeof(OPTIONS_DEFAULT_PATH) - 1];
+        JLI_Snprintf(optionsfile, sizeof(optionsfile), "%s" LIB_PATH LIBARCHNAME OPTIONS_DEFAULT_PATH, jrepath);
```

Comment:
Did you consider folding `LIB_PATH` and `LIBARCHNAME` into `OPTIONS_DEFAULT_PATH`?
```c
#define OPTIONS_DEFAULT_PATH LIB_PATH LIBARCHNAME "/default/options.default"
        char optionsfile[sizeof(jrepath) + sizeof(OPTIONS_DEFAULT_PATH) - 1];
        JLI_Snprintf(optionsfile, sizeof(optionsfile), "%s" OPTIONS_DEFAULT_PATH, jrepath);
```

Reply:
Good idea, updated.